```
BEGIN ASSIGNMENT
requirements: ../../requirements.txt
overwrite_requirements: true
solutions_pdf: true
generate:
    show_stdout: true
    seed: 42
files:
    - earnings.csv
    - regression.png
plugins:
    - otter.plugins.builtin.GoogleSheetsGradeOverride:
        credentials_json_path: ../../sheets-api-credentials.json
        sheet_url: https://docs.google.com/spreadsheets/d/1WhPrOz4x18CABYVTHgbThSZgjSL07fwZx4_mOJWs4Ic/edit
```

<table style="width: 100%;" id="nb-header">
    <tr style="background-color: transparent;"><td>
        <img src="https://data-88e.github.io/assets/images/blue_text.png" width="250px" style="margin-left: 0;" />
    </td><td>
        <p style="text-align: right; font-size: 10pt;"><strong>Economic Models</strong>, Fall 2022<br>
            Dr. Eric Van Dusen <br>
        Notebook by Vaidehi Bulusu <br>
</table>

# Project 4: Econometrics

In the textbook and this week's lecture, we gave you a theoretical introduction to econometrics. In this project, you'll get the chance to apply what you've learned and see how econometrics is actually used by economists!

This project is based on an influential study on the relationship between a person's height and labor market outcomes, and is divided into 3 sections:

1. Simple Linear Regression

2. Multiple Linear Regression

3. Reading Econometrics Tables

You can refer to the [*Econometrics*](https://data-88e.github.io/textbook/content/11-econometrics/index.html) chapter in the textbook and week 12 resources for help on this project.

In [1]:
from datascience import *
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
import warnings
warnings.simplefilter(action='ignore')
plt.style.use('seaborn-muted')
%matplotlib inline
plt.rcParams["figure.figsize"] = [10,7]

 ## Part 1: Simple Linear Regression

Several studies have identified a positive correlation between a person's height and labor market outcomes: on average, taller people have jobs that are of a higher status and pay more. In their paper, *[Stature and Status:
Height, Ability, and Labor Market Outcomes](https://www.nber.org/system/files/working_papers/w12466/w12466.pdf)* (2008), economists Anne Case and Christina Paxson analyze the data from the US National Health Interview Survey in 1994 to explain this association. This is the data we will be using for our analysis.

In the first part of the project, we will use simple (bivariate) linear regression to look at the association between a person's height and earnings, and consider the problems with limiting our regression model to just 1 regressor.

We start by importing the `earnings.csv` dataset which contains information about the characteristics and labor market outcomes for 17,870 workers.

In [2]:
earnings = Table().read_table("earnings.csv")
earnings.show(5)

<div class="alert alert-warning">
    
Before proceeding, please read the <a href="https://wps.pearsoned.com/wps/media/objects/11422/11696965/data3eu/Earnings_and_Height_Description.pdf" target="_blank">data description</a> for this study which gives you information about each variable. **This is very important.**
    
</div>

**Question 1.1:** What do you expect the sign of the relationship between height and earnings to be? Explain your answer.

```
BEGIN QUESTION
name: q1_1
manual: true
```

<div class="alert alert-danger">
    
<strong>SOLUTION:</strong> Can award credit to any reasonable response. For example, there may be a positive association because taller people may be perceived as more intelligent so they earn more.

</div>

Note: We generally take the log of earnings in regression models but we’re not doing that in this project for the sake of simplicity. We’ll be using log earnings as the dependent variable in section 3.

**Question 1.2:** Here's a simple linear regression equation to model the association between height and earnings:

$$\text{Earnings} = \beta_1 \times \text{Height} + \alpha$$

Perform a regression of `earnings` on `height`. Don't forget to add a constant term!

```
BEGIN QUESTION
name: q1_2
```

In [3]:
""" # BEGIN PROMPT
y_1_2 = ...
x_1_2 = ...
model_1_2 = ...
results_1_2 = ...
""" # END PROMPT
# BEGIN SOLUTION NO PROMPT
y_1_2 = earnings.column("earnings")
x_1_2 = earnings.select("height").to_df()
model_1_2 = sm.OLS(y_1_2, sm.add_constant(x_1_2))
results_1_2 = model_1_2.fit()
# END SOLUTION
results_1_2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.011
Model:                            OLS   Adj. R-squared:                  0.011
Method:                 Least Squares   F-statistic:                     196.5
Date:                Sun, 13 Nov 2022   Prob (F-statistic):           2.13e-44
Time:                        22:59:25   Log-Likelihood:            -2.0755e+05
No. Observations:               17870   AIC:                         4.151e+05
Df Residuals:                   17868   BIC:                         4.151e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       -512.7336   3386.856     -0.151      0.880   -7151.299    6125.832
height       707.6716     50.489     14.016      0.000     608.708     806.635
==============================================================================
Omnibus:                   346151.826   Durbin-Watson:                   1.683
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             1827.913
Skew:                           0.397   Prob(JB):                         0.00
Kurtosis:                       1.649   Cond. No.                     1.13e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.13e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [4]:
## TEST ##
len(results_1_2.params) == 2

True

In [5]:
## TEST ##
np.count_nonzero(y_1_2) == 17870

True

In [6]:
## TEST ##
100 < np.sum(results_1_2.params) < 200

True

In [7]:
## HIDDEN TEST ##
707.672 in set(np.round(results_1_2.params, 3))

True

**Question 1.3:**
Why should we include a constant term in this regression model?

<ol type="A" style="list-style-type: lower-alpha;">
    <li>Because we expect the slope of the line of best fit may be zero.  </li>
    <li>Because we expect the slope of the line of best fit may be non-zero.  </li>
    <li>Because we expect the x-intercept of the line of best fit may be non-zero.  </li>
    <li>Because we expect the y-intercept of the line of best fit may be non-zero.  </li>
</ol>

Assign a letter corresponding to your answer to `q1_3` below. For example, `q1_3 = 'a'`.

```
BEGIN QUESTION
name: q1_3
points: 
    - 0
    - 1
```

In [8]:
q1_3 = 'd' # SOLUTION

In [9]:
## TEST ##
q1_3 in ['a', 'b', 'c', 'd']

True

In [10]:
## HIDDEN TEST ##
q1_3 == 'd'

True

**Question 1.4:** What is the estimated association between `earnings` and `height`?

```
BEGIN QUESTION
name: q1_4
```

In [11]:
""" # BEGIN PROMPT
result_1_4 = ...
""" # END PROMPT
# BEGIN SOLUTION NO PROMPT
result_1_4 = results_1_2.params['height']
# END SOLUTION
result_1_4

707.67155842741204

In [12]:
## TEST ##
type(result_1_4) == float

False

In [13]:
## HIDDEN TEST ##
np.isclose(result_1_4, 707.67155842744069)

True

**Question 1.5:** Is the association statistically significant? Answer `True` or `False`.

```
BEGIN QUESTION
name: q1_5
```

In [14]:
""" # BEGIN PROMPT
result_1_5 = ...
""" # END PROMPT
# BEGIN SOLUTION NO PROMPT
result_1_5 = True
# END SOLUTION
result_1_5

True

In [15]:
## TEST ##
type(result_1_5) == bool

True

In [16]:
## HIDDEN TEST ##
result_1_5

True

**Question 1.6:** Interpret the slope on the `height` variable (including the units). 

<ol type="A" style="list-style-type: lower-alpha;">
    <li>A 1 inch increase in height corresponds to around \$707.7 increase in earnings  </li>
    <li>A 1 inch increase in height corresponds to around \$707.7 dncrease in earnings  </li>
    <li>A 1% increase in height corresponds to around \$707.7 increase in earnings  </li>
    <li>A 1% increase in height corresponds to around \$707.7 decrease in earnings  </li>
</ol>

Assign a letter corresponding to your answer to `q1_6` below. For example, `q1_6 = 'a'`.

```
BEGIN QUESTION
name: q1_6
points: 
    - 0
    - 1
```

In [17]:
q1_6 = 'a' # SOLUTION

In [18]:
## TEST ##
q1_6 in ['a', 'b', 'c', 'd']

True

In [19]:
## HIDDEN TEST ##
q1_6 == 'a'

True

**Question 1.7:** Interpret the intercept of the regression (including the units). Does this make practical sense? 

<ol type="A" style="list-style-type: lower-alpha;">
    <li>When height is 0, earnings are estimated to be around -\$512.7.  </li>
    <li>When height is 0, earnings are estimated to be around \$512.7.  </li>
    <li>When earnings are 0, height are estimated to be around -512.7 inches.  </li>
    <li>When earnings are 0, height are estimated to be around 512.7 inches.  </li>
</ol>

Assign a letter corresponding to your answer to `q1_7` below. For example, `q1_7 = 'a'`.

```
BEGIN QUESTION
name: q1_7
points: 
    - 0
    - 1
```

In [20]:
q1_7 = 'a' # SOLUTION

In [21]:
## TEST ##
q1_7 in ['a', 'b', 'c', 'd']

True

In [22]:
## HIDDEN TEST ##
q1_7 == 'a'

True

**Question 1.8:** If the slope on the independent variable is statistically significant, can we infer a causal relationship between the two variables? Why or why not? When can you infer a causal relationship between 2 variables based on the results of the study?

*Hint: Think about the type of study we're looking at.*

```
BEGIN QUESTION
name: q1_8
manual: true
```

<div class="alert alert-danger">
    
<strong>SOLUTION:</strong> No, a statistically significant regression slope is not enough to infer a causal relationship between the independent and dependent variables. This is because of the presence of omitted variables (an example in this case is the number of years of education). You can infer a causal relationship only from randomized controlled trials.

</div>

**Question 1.9:** Use the slope and intercept from the regression in question 1.2 to generate predictions for `earnings`.

```
BEGIN QUESTION
name: q1_9
points: 
    - 0
    - 0.5
    - 0.5
```

In [23]:
""" # BEGIN PROMPT
predictions_1_9 = ...
""" # END PROMPT
# BEGIN SOLUTION NO PROMPT
predictions_1_9 = (earnings.column("height") * results_1_2.params[1]) + results_1_2.params[0]
# END SOLUTION
predictions_1_9

array([ 45485.91770787,  45485.91770787,  41947.55991573, ...,
        45485.91770787,  47608.93238315,  47608.93238315])

In [24]:
## TEST ##
len(predictions_1_9) == 17870

True

In [25]:
## HIDDEN TEST ##
np.isclose(np.mean(predictions_1_9), 46875.322107839529)

True

In [26]:
## HIDDEN TEST ##
np.isclose(predictions_1_9[0], 45485.918)

True

**Question 1.10:** Calculate the RMSE for your regression predictions from question 1.9.

```
BEGIN QUESTION
name: q1_10
```

In [27]:
""" # BEGIN PROMPT
rmse_1_10 = ...
""" # END PROMPT
# BEGIN SOLUTION NO PROMPT
rmse_1_10 = (np.mean((earnings.column("earnings") - predictions_1_9)**2))**0.5
# END SOLUTION
rmse_1_10

26775.741156313969

In [28]:
## TEST ##
10000 < rmse_1_10 < 50000

True

In [29]:
## HIDDEN TEST ##
np.isclose(rmse_1_10, 26775.741)

True

**Question 1.11:** Which one of the following is true about the RMSE? 

<ol type="A" style="list-style-type: lower-alpha;">
    <li>RMSE is the total sum of square error in the regression.   </li>
    <li>RMSE means that on average the predicted earnings are off by $26,775.7.  </li>
    <li>It is possible for the RMSE to increase if we add unrelated variables to the regression.  </li>
    <li>Higher RMSE means the model more accurately predicts the dependent variable.  </li>
</ol>

Assign a letter corresponding to your answer to `q1_11` below. For example, `q1_11 = 'a'`.

```
BEGIN QUESTION
name: q1_11
points: 
    - 0
    - 1
```

In [30]:
q1_11 = 'b' # SOLUTION

In [31]:
## TEST ##
q1_11 in ['a', 'b', 'c', 'd']

True

In [32]:
## HIDDEN TEST ##
q1_11 == 'b'

True

## Part 2: Multiple Linear Regression

Now, let's perform multiple linear regression to account for potential confounding variables in our model. For simplicity, we will be using only the following additional regressors: `age`, `educ`, `sex` and `weight`.

**Question 2.1:** Pick 2 variables from the 4 additional variables we will be incorporating in the model (`age`, `educ`, `sex` and `weight`) that you think might be potential confounders in the study. Provide a brief explanation for why you think each of the variables may be a confounder. 

*Hint: Recall that a confounder is related to both the independent and dependent variables, so in your answer, make sure to talk about how each confounder is related to `height` and `weight`.*

```
BEGIN QUESTION
name: q2_1
manual: true
```

<div class="alert alert-danger">
    
<strong>SOLUTION:</strong> Can award credit to any reasonable response (e.g. `educ` may be positively correlated with both `height` and `earnings` so it's a potential confounding variable).

</div>

**Question 2.2:** Perform another regression with the following new regressors: `age`, `educ`, `sex` and `weight` (also include `height`).

```
BEGIN QUESTION
name: q2_2
```

In [33]:
""" # BEGIN PROMPT
y_2_2 = ...
x_2_2 = ...
model_2_2 = ...
results_2_2 = ...
""" # END PROMPT
# BEGIN SOLUTION NO PROMPT
y_2_2 = earnings.column("earnings")
x_2_2 = earnings.select("height", "age","educ", "sex", "weight").to_df()
model_2_2 = sm.OLS(y_2_2, sm.add_constant(x_2_2))
results_2_2 = model_2_2.fit()
# END SOLUTION
results_2_2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.170
Model:                            OLS   Adj. R-squared:                  0.169
Method:                 Least Squares   F-statistic:                     729.7
Date:                Sun, 13 Nov 2022   Prob (F-statistic):               0.00
Time:                        22:59:26   Log-Likelihood:            -2.0598e+05
No. Observations:               17870   AIC:                         4.120e+05
Df Residuals:                   17864   BIC:                         4.120e+05
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -4.745e+04   4348.852    -10.911      0.000    -5.6e+04   -3.89e+04
height       414.7890     68.097      6.091      0.000     281.312     548.265
age          335.0852     18.420     18.191      0.000     298.980     371.190
educ        3948.2298     70.529     55.980      0.000    3809.986    4086.473
sex          586.9203    520.681      1.127      0.260    -433.665    1607.505
weight        -5.0913      3.878     -1.313      0.189     -12.692       2.509
==============================================================================
Omnibus:                     3334.172   Durbin-Watson:                   1.908
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             1045.069
Skew:                           0.362   Prob(JB):                    1.16e-227
Kurtosis:                       2.062   Cond. No.                     4.61e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.61e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [34]:
## TEST ##
len(results_2_2.params) == 6

True

In [35]:
## TEST ##
4000 < np.sum(results_2_2.params[1:]) < 8000

True

In [36]:
## HIDDEN TEST ##
335.0852 in set(np.round(results_2_2.params, 4)) # coef on age

True

In [37]:
## HIDDEN TEST ##
3948.2298 in set(np.round(results_2_2.params, 4)) # coef on educ

True

**Question 2.3:** Compare the coefficient on `height` from the regression model in question 2.2 to the coefficient on `height` from the regression model in question 1.2. What does this tell you about the nature of the omitted variable bias in the previous model (is it positive or negative)?

Fill in the blanks: The coefficient in 2.2 is \_\_\_\_\_ which means that the omitted variable bias in 1.2 is overall \_\_\_\_\_. 

<ol type="A" style="list-style-type: lower-alpha;">
    <li>higher; positive</li>
    <li>lower; positive</li>
    <li>higher; negative</li>
    <li>lower; negative</li>
</ol>

Assign a letter corresponding to your answer to `q2_3` below. For example, `q2_3 = 'a'`.

```
BEGIN QUESTION
name: q2_3
points: 
    - 0
    - 1
```

In [38]:
q2_3 = 'b' # SOLUTION

In [39]:
## TEST ##
q2_3 in ['a', 'b', 'c', 'd']

True

In [40]:
## HIDDEN TEST ##
q2_3 == 'b'

True

**Question 2.4:** Look at the regression coefficients for the additional variables we included in question 2.2. Do the coefficients align with your intuition from 2.1 (in terms how they are related to earnings)?

```
BEGIN QUESTION
name: q2_4
manual: true
```

<div class="alert alert-danger">
    
<strong>SOLUTION:</strong> Should talk about the sign of the coefficients of the variables they chose in 2.1.

</div>

**Question 2.5:** Let's consider the effect of education on earnings. According to the result in 2.2, is the level of education correlated with earnings? Does adding this variable help alleviate omitted variable bias? Why or why not? Explain your answers. 

```
BEGIN QUESTION
name: q2_5
manual: true
```

<div class="alert alert-danger">
    
<strong>SOLUTION:</strong> Yes, it seems that the level of education is positively correlated with earnings--for every 1 additional year of schooling, the annual earnings on average will increase by \$3948.23. It helps alleviate omitted variable bias, because education is an important predicator of earnings and therefore should be included in the model. (Alternatively, one can also argue that it doesn't help alleviate omitted variable bias because the level of education may not be directly correlated with a person's height, so adding education won't bring down the coefficient of height)

</div>

**Question 2.6:** If we computed the RMSE for this new regression model, do you think it would be higher or lower than the RMSE we computed in 1.10? 

<ol type="A" style="list-style-type: lower-alpha;">
    <li>Lower</li>
    <li>Higher</li>
    <li>It depends on the specific choice of variables </li>
</ol>

Assign a letter corresponding to your answer to `q2_6` below. For example, `q2_6 = 'a'`.

```
BEGIN QUESTION
name: q2_6
points: 
    - 0
    - 1
```

In [41]:
q2_6 = 'a' # SOLUTION

In [42]:
## TEST ##
q2_6 in ['a', 'b', 'c']

True

In [43]:
## HIDDEN TEST ##
q2_6 == 'a'

True

**Question 2.7:** Now that we've accounted for some additional confounding variables do you think it makes sense for us to infer a causal relationship between height and earnings?

<ol type="A" style="list-style-type: lower-alpha;">
    <li>Yes, because the coefficient for height is now highly significant. </li>
    <li>Yes, because we have eliminated omitted variable bias by adding control variables in 2.2. </li>
    <li>No, because there may be other omitted variables we haven't accounted for.</li>
    <li>No, because the set of control variables added in 2.2 is a poor choice. </li>
</ol>

Assign a letter corresponding to your answer to `q2_7` below. For example, `q2_7 = 'a'`.

```
BEGIN QUESTION
name: q2_7
points: 
    - 0
    - 1
```

In [44]:
q2_7 = 'c' # SOLUTION

In [45]:
## TEST ##
q2_7 in ['a', 'b', 'c', 'd']

True

In [46]:
## HIDDEN TEST ##
q2_7 == 'c'

True

**Question 2.8:** Using regression results in 2.2, now let’s try to predict a person’s earnings based on their characteristics.

What would the predicted earnings of a 35-year-old woman with a Bachelor's degree who is 64 inches tall and 124 pounds?

*Hint: A person with a Bachelor's degree has completed 15 years of education.*

```
BEGIN QUESTION
name: q2_8
```

In [47]:
""" # BEGIN PROMPT
prediction_2_8 = ...
""" # END PROMPT
# BEGIN SOLUTION NO PROMPT
prediction_2_8 = results_2_2.params[0] + (results_2_2.params[1]*64) + (results_2_2.params[2]*35) + (results_2_2.params[3]*15) + (results_2_2.params[4]*0) + (results_2_2.params[5]*124)
# END SOLUTION
prediction_2_8

49414.269967442197

In [48]:
## TEST ##
20000 < prediction_2_8 < 50000

True

In [49]:
## HIDDEN TEST ##
np.isclose(prediction_2_8, 49414.27)

True

**Question 2.9:** Say you wanted to know the relationship between gender (`sex`: a binary variable) and income (`earnings`: a continuous variable). Based on our regression results, how is gender correlated with income?

*Hint*: Think about what it means when the coefficient of `sex` is 0 and 1. Also, you can try changing your input for `sex` in 2.8, and see what happens.

Assuming all else equal, ...

<ol type="A" style="list-style-type: lower-alpha;">
    <li>On average, male earns \$586.92 more than female. </li>
    <li>On average, female earns \$586.92 more than male. </li>
    <li>For every 1 inch increase in height, male's income will increase on average \$586.92 more than that of female. </li>
    <li>For every 1 inch increase in height, female's income will increase on average \$586.92 more than that of male. </li>
</ol>

Assign a letter corresponding to your answer to `q2_9` below. For example, `q2_9 = 'a'`.

```
BEGIN QUESTION
name: q2_9
points: 
    - 0
    - 1
```

In [50]:
q2_9 = 'a' # SOLUTION

In [51]:
## TEST ##
q2_9 in ['a', 'b', 'c', 'd']

True

In [52]:
## HIDDEN TEST ##
q2_9 == 'a'

True

## Part 3: Reading Econometrics Tables

Researchers tend to run multiple regression models which they summarize in econometrics tables.

Below is a table taken from the paper. It shows the regression results from 2 studies on the relationship between height and earnings: the British Cohort Study (BCS) and National Child Development Study (NCDS).

<img src = "https://i.imgur.com/a2o9OPA.png">

<div class="alert alert-warning">
    
Make sure to read the table (including the note at the bottom) before proceeding.
    
</div>

Note that for the questions below, **log earnings** is the dependent variable. Recall the [interpretation of the slope](https://data-88e.github.io/textbook/content/01-demand/03-log-log.html) in this case.

**Question 3.1:** According to the table, what did the British Cohort Study (1970) find about the relationship between height at age 30, test scores ages 5 and 10, and earnings for women? Use the results with extended controls added in. Which of the followings are correct? There can be 1-4 correct answers. 

<ol type="A" style="list-style-type: lower-alpha;">
    <li>The coefficient on height is statistically significant. </li>
    <li>The coefficient on height means for every 1 inch increase in height at age 30, the annual earnings will on average increase by 0.002 dollars. </li>
    <li>The coefficient on test scores is statistically significant. </li>
    <li>The coefficient on test scores means for every 1 point increase in test scores ages 5 and 10, the annual earnings will on average increase by 19.75 dollars. </li>
</ol>

Assign an array of letters corresponding to your answer to `q3_1` below. For example, `q3_1 = make_array('a', 'b', 'c', 'd')`.

```
BEGIN QUESTION
name: q3_1
points: 
    - 0
    - 0
    - 1
```

In [53]:
q3_1 = make_array('c') # SOLUTION

In [54]:
## TEST ##
len(q3_1) > 0 

True

In [55]:
## TEST ##
np.all([x in ['a', 'b', 'c', 'd'] for x in q3_1])

True

In [56]:
## HIDDEN TEST ##
len(q3_1) == 1 and q3_1[0] == 'c'  # b and d are incorrect because the dependent variable is *log* annual earnings

True

**Question 3.2:** Now look at the results of the British Cohort Study for men. There are 3 coefficients reported for height at age 30. Give a brief interpretation of each of these coefficients (including the units). In your explanation, talk about what is causing the differences among these coefficients (hint: look at the *Test scores* and *Extended controls* rows).

Note: Assume height is measured in inches for this study.

```
BEGIN QUESTION
name: q3_2
manual: true
```

<div class="alert alert-danger">
    
<strong>SOLUTION:</strong> Each coefficient: % change in **annual** earnings per 1 inch change in height (as the dependent variable is log earnings) for 30-year-old men. For the first coefficient, they are not accounting for test scores or have extended controls. For the second coefficient, they are accounting for test scores but don't have extended controls. For the third coefficient, they are accounting for test scores + have extended controls. This is what is causing the difference.

</div>

## Conclusion

This brings us to the end of project 4! You've learned some key econometrics skills such as running regressions and reading econometrics tables. You've also developed an intuition for ordinary linear regression, omitted variable bias and regression with dummy variables.

Also, we didn't cover a large part of Case and Paxson's [fascinating study](https://www.nber.org/system/files/working_papers/w12466/w12466.pdf) so if you're interested in how they explain the positive association between height and earnings, we recommend giving the paper a read!

Congratulations on finishing your final DATA 88E project!

---